components/script/stylesheet_loader.rs

```diff
@@ -213,6 +202,187 @@ impl FetchResponseListener for StylesheetContext {
+                        Some(Arc::new(Stylesheet::from_bytes(
+                            &data,
+                            final_url,
+                            protocol_encoding_label,
+                            Some(environment_encoding),
+                            Origin::Author,
+                            media.take().unwrap(),
+                            shared_lock,
+                            Some(&loader),
+                            // No error reporting in async parse mode.
+                            None,
+                            quirks_mode,
+                        )))
+                    },
+                    StylesheetSource::Import(ref stylesheet) => {
+                        Stylesheet::update_from_bytes(
+                            &stylesheet,
+                            &data,
+                            protocol_encoding_label,
+                            Some(environment_encoding),
+                            final_url,
+                            Some(&loader),
+                            // No error reporting in async parse mode.
+                            None,
+                        );
+                        // We already have a stylesheet for the import rule,
+                        // so no need to create and return a new one.
+                        None
+                    },
+                };
+
+                // TODO(mandreyel): What to do with errors?
+                let _ = script_chan.send(Msg::Common(CommonScriptMsg::Task(
+                    ScriptThreadEventCategory::StylesheetLoad,
+                    Box::new(FinishAsyncStylesheetLoadTask {
+                        successful,
+                        origin_clean,
+                        top_level_stylesheet: sheet,
+                        elem: trusted_elem,
+                        document,
+                        url,
+                        request_generation_id,
+                    }),
+                    pipeline_id,
+                    TaskSourceName::Networking,
+                )));
+            });
+        } else {
+            // The load was not successful, let document know.
+            let owner = elem
+                .upcast::<Element>()
+                .as_stylesheet_owner()
+                .expect("Stylesheet not loaded by <style> or <link> element!");
+            owner.set_origin_clean(self.origin_clean);
+            if owner.parser_inserted() {
+                document.decrement_script_blocking_stylesheet_count();
+            }
+
+            document.finish_load(LoadType::Stylesheet(self.url.clone()));
+
+            let successful = false;
+            if let Some(any_failed) = owner.load_finished(successful) {
+                let event = if any_failed {
+                    atom!("error")
+                } else {
+                    atom!("load")
+                };
+                elem.upcast::<EventTarget>().fire_event(event);
```

Comment:
Since this branch is only executed on fetch failure, can we unconditionally fire an `error` event here, @emilio?